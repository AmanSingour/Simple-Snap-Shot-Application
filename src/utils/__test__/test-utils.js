import React from "react";
import { Provider } from "react-redux";
import { createStore } from "redux";
import reducers from "../../services/reducers";

import { render as rtlRender } from "@testing-library/react";

// REDUX PROVIDER SETUP

function render(
  ui,
  {
    initialState,
    store = createStore(reducers, initialState),
    ...renderOptions
  } = {}
) {

  function Wrapper({ children }) {
    return <Provider store={store}>{children}</Provider>;
  }
  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
}

export * from "@testing-library/react";

export { render };
