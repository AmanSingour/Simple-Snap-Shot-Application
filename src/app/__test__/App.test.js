///import { expect } from '@jest/globals'
import { render } from "../../utils/__test__/test-utils";
import App from "./../index";

describe("***App running status***", () => {
  let wrapper;

  beforeEach(() => (wrapper = render(<App />, { initialState: {} })));

  it("renders without crushing", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("App exist", () => {
    const { getByTestId } = wrapper;
    expect(getByTestId("app")).toBeInTheDocument();
  });
});
