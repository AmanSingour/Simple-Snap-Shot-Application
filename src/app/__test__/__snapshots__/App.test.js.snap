// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`***App running status*** renders without crushing 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="App"
        data-testid="app"
      >
        <div
          class="ui container"
        >
          <div
            class="Container"
          >
            <form
              class="ui form Form"
            >
              <h2
                class="ui header"
              >
                Login
              </h2>
              <br />
              
              <div
                class="field InputField"
              >
                <div
                  class="required field"
                >
                  <div
                    class="ui left icon input"
                  >
                    <i
                      aria-hidden="true"
                      class="at icon"
                    />
                    <input
                      name="email"
                      placeholder="Username or Email"
                      required=""
                      type="email"
                    />
                  </div>
                </div>
              </div>
              <div
                class="field InputField"
              >
                <div
                  class="required field"
                >
                  <div
                    class="ui left icon input"
                  >
                    <i
                      aria-hidden="true"
                      class="key icon"
                    />
                    <input
                      name="password"
                      placeholder="Password"
                      required=""
                      type="password"
                    />
                  </div>
                </div>
              </div>
              <div
                class="field Button"
              >
                <button
                  class="ui positive disabled button"
                  disabled=""
                  tabindex="-1"
                  type="submit"
                >
                  LOGIN
                </button>
              </div>
              <label>
                Don't have an account?
                <a
                  href="#/signup"
                >
                  Signup
                </a>
              </label>
            </form>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="App"
      data-testid="app"
    >
      <div
        class="ui container"
      >
        <div
          class="Container"
        >
          <form
            class="ui form Form"
          >
            <h2
              class="ui header"
            >
              Login
            </h2>
            <br />
            
            <div
              class="field InputField"
            >
              <div
                class="required field"
              >
                <div
                  class="ui left icon input"
                >
                  <i
                    aria-hidden="true"
                    class="at icon"
                  />
                  <input
                    name="email"
                    placeholder="Username or Email"
                    required=""
                    type="email"
                  />
                </div>
              </div>
            </div>
            <div
              class="field InputField"
            >
              <div
                class="required field"
              >
                <div
                  class="ui left icon input"
                >
                  <i
                    aria-hidden="true"
                    class="key icon"
                  />
                  <input
                    name="password"
                    placeholder="Password"
                    required=""
                    type="password"
                  />
                </div>
              </div>
            </div>
            <div
              class="field Button"
            >
              <button
                class="ui positive disabled button"
                disabled=""
                tabindex="-1"
                type="submit"
              >
                LOGIN
              </button>
            </div>
            <label>
              Don't have an account?
              <a
                href="#/signup"
              >
                Signup
              </a>
            </label>
          </form>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
